Hoxtygen/home360-server
# Ref: refactor listing entity move validation to listing dto (HOM-127)

## What does this PR do?
* Refactors listing entity and listing DTO
* Refactors service and controller layers for new listing creation

## Description of tasks to be completed
* Modify listing entity
* Modify listing DTO
* Update logic for creating new listing in service layer
*  Update logic for handling  new listing creation in controller layer
* Add handler for `HttpMessageNotReadableException` in global exception handler
* Format files with spotless

## How should this be manually tested?
* Clone the repository `git clone <repo_url>`
* `cd` into the project directory and install dependencies
* Run the application
* Create a new user, verify email and login.
* Copy the access token from login response
* Send a `POST` request to create new listing
```
curl --location 'http://localhost:8080/api/v1/listings' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer ${access token}' \
--data '{
    "title": "2-bedroom flat at Abobaku, Egbejila",
    "description": "Standard newly built two bedrooms apartment with full pop ceiling and all modern facilities is available at Egbejila",
    "furnishing": "Fully tiled, borehole water, wardrobe, kitchen cabinets, floored compound, floored and gated.",
    "position": "Inside Federaal Housing Estate",
    "address": {
        "state": "Kwara",
        "city": "Ilorin",
        "houseNumber": "1",
        "streetName": "Abobaku, Egbejila",
        "lga": "Ilorin West"
    },
    "miscellaneous": "Lorem ipsum dolor sit amet consectetur adipscing",
    "availableFrom": "2024-11-01T15:31",
    "cost": {
        "annualRent": 350000,
        "agentFee": 50000,
        "cautionFee": 30000

    },
    "facilityQuality": "normal",
    "petsAllowed": "YES",
    "details": ["Very good house with electricity", "upstairs"],
    "applicationDocs": ["Proof of income", "Electricity bill", "Letter from last agent you worked with or landlord"],
    "apartmentInfo": {
        "roomNums": 3,
        "bathroomNums": 2,
        "bedroomNums": 2,
        "apartmentType": "APARTMENT"
    },
    "apartmentImages": ["image_url", "image_url", ]
}'

```


## What are the relevant Jira board stories?
[HOM-127](https://wasiu-dev.atlassian.net/jira/software/projects/HOM/boards/2?selectedIssue=HOM-127)
Ref #3 

[HOM-127]: https://wasiu-dev.atlassian.net/browse/HOM-127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
